### PR TITLE
feat: show patch pump icon instead of cannula

### PR DESCRIPTION
**Pull request by Andries-Smit**
Omnipod, does not have a cannula, so let us replace the icon when a patch pump is used (Dash and Eros). Both in the status lights in the overview and in the action Careportal section.

Image

image

Note: Omnipod Dash does not use a link or has replaceable batteries, therefore the battery can be section hidden. There will be a separate PR for this after #1200 is merged

### CODE REVIEW

**Comment by MilosKozak** [comment]
there will be more patch pumps available soon. Could you make the code more general?
ie. add to to `PumpInterface` `val isPatchPump: Boolean`, extend `PumpDescription` to provide this info and rename variables used in this code (Omnipod -> Patch)?

**Comment by Andries-Smit** [comment]
@MilosKozak the more, the merrier! will add it

**Comment by Andries-Smit** [comment]
Refactored to be a more generic patch pump feature instead of Omnipod only

File: app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.kt [hunk 1]
```diff
@@ -823,7 +823,15 @@ class OverviewFragment : DaggerFragment(), View.OnClickListener, OnLongClickList
...
         binding.infoLayout.time.text = dateUtil.timeString(dateUtil.now())
         // Status lights
-        binding.statusLightsLayout.statusLights.visibility = (sp.getBoolean(R.string.key_show_statuslights, true) || config.NSCLIENT).toVisibility()
+        val isPatchPump = activePlugin.activePump.pumpDescription.isPatchPump
+        binding.statusLightsLayout.apply {
+            cannulaOrPatch.setImageResource(if (isPatchPump) R.drawable.ic_patch_pump_outline else R.drawable.ic_cp_age_cannula)
+            cannulaOrPatch.contentDescription = rh.gs(if (isPatchPump) R.string.statuslights_patch_pump_age else R.string.statuslights_cannula_age)
+            cannulaOrPatch.scaleX = if (isPatchPump) 1.4f else 2f
+            cannulaOrPatch.scaleY = cannulaOrPatch.scaleX
+            insulinAge.visibility = (!isPatchPump).toVisibility()
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.kt:832]: kotlin notation for this is `isPatchPump.not().toVisibility()`

Comment (Andries-Smit) [reply app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.kt:832]: nice one, that looks much better, I start to appreciate Kotlin more and more

**Pull request merged**